### DISCUSSION TITLE
Getting the command dispatcher

### DISCUSSION THREAD

**Discussion by miraisoft-de** [FabricMC/fabric-api#2453] [Mod Dev Support] [answered]
In Minecraft 1.18.2 you could get the command dispatcher with the line
`CommandDispatcher<FabricClientCommandSource> commandDispatcher = ClientCommandManager.DISPATCHER;`
It works as expected.

For Minecraft 1.19.x I changed it to 
`CommandDispatcher<FabricClientCommandSource> commandDispatcher = ClientCommandManager.getActiveDispatcher();`
Now the variable is just `null`. 
What can I do to get the dispatcher in MC 1.19.x?

**Comment by warjort** [comment, marked as answer]
The javadoc says it can be null:
https://github.com/FabricMC/fabric/blob/7454701c678c551a841a58847fdab5473fc4690b/fabric-command-api-v2/src/client/java/net/fabricmc/fabric/api/client/command/v2/ClientCommandManager.java#L73

**Reply by miraisoft-de** [reply]
Thanks. I've just found the answer myself.
It is explained in the javadoc of `ClientCommandManager`
and also at [https://fabricmc.net/wiki/tutorial:commands](https://fabricmc.net/wiki/tutorial:commands)
